### DISCUSSION TITLE
Opinions on #1044

### DISCUSSION THREAD

**Discussion by ava-fred** [eclipse/lsp4e#1093] [General]
This discussion is to solicit opinions on #1044 from the community. 
The PR changes the fact that, if LanguageServerWrapper.start() failed, the consumer only saw a CancellationException and not the exception that actually caused the failure. As a consequence of this change, calls to LanguageServer.start() now don't do anything after a failure. If retrying the startup after a failure with the same LanguageServerWrapper object makes sense for some reason, then this has to be done explicitly by calling restart().
I discussed this change with @rubenporras , and he agrees that it makes sense. However, we both realise that the change in behavior is significant, and welcome other people's opinions on it.

**Comment by rubenporras** [comment]
If there are no opinions by the end of the week, I would like to merge #1044
